examples/morpheusvm/auth/bls.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package auth
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/crypto"
+	"github.com/ava-labs/hypersdk/crypto/bls"
+	"github.com/ava-labs/hypersdk/examples/morpheusvm/consts"
+	"github.com/ava-labs/hypersdk/examples/morpheusvm/storage"
+	"github.com/ava-labs/hypersdk/state"
+	"github.com/ava-labs/hypersdk/utils"
+)
+
+var _ chain.Auth = (*BLS)(nil)
+
+const (
+	BLSComputeUnits = 10
+	BLSSize         = bls.PublicKeyLen + bls.SignatureLen
+)
+
+type BLS struct {
+	Signer    bls.PublicKey `json:"signer"`
```

Comment:
If the AvalancheGo library operates over pointers, I would recommend using them here instead of dereferencing things.